### PR TITLE
Use `protobuf/jsonpb` instead of `golang json` in some cases

### LINKED ISSUE #154: We should use github.com/golang/protobuf/jsonpb instead of golang json package.
Currently if a container exit with exit code 0, when we do `crictl inspect`, there won't be exit code field, because that is omitted https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/apis/cri/v1alpha1/runtime/api.pb.go#L2119.

We should use `github.com/golang/protobuf/jsonpb`, so that we could configure `EmitDefaults` to fix this.

### PR DESCRIPTION
**Pull request by miaoyq**
For #154 

/cc @Random-Liu 

Signed-off-by: Yanqiang Miao <miao.yanqiang@zte.com.cn>

### CODE REVIEW

File: vendor/github.com/golang/protobuf/jsonpb/jsonpb.go [hunk 1]
```diff
@@ -0,0 +1,829 @@
+// Go support for Protocol Buffers - Google's data interchange format
```
Comment (feiskyer) [inline vendor/github.com/golang/protobuf/jsonpb/jsonpb.go]: This is a vendor update, does vendor.conf need update?

**Comment by Random-Liu** [comment]
@miaoyq How do we handle extra information then?

File: vendor/github.com/golang/protobuf/jsonpb/jsonpb.go (hunk 1 above)
Comment (miaoyq) [reply vendor/github.com/golang/protobuf/jsonpb/jsonpb.go]: Only update jsonpb here, I think we dont need update vendor.conf unless other model depend on the new version of protobuf.  :)

**Comment by miaoyq** [comment]
@Random-Liu We could handle extra info via "outputStatusInfo"

**Comment by miaoyq** [comment]
@Random-Liu "protobufObjectToJSON(status)" handles resp.Status, and "json.Marshal" handles resp.Info

**Comment by feiskyer** [comment]
> "protobufObjectToJSON(status)" handles resp.Status, and "json.Marshal" handles resp.Info

LGTM. @Random-Liu do you have any other concerns?

**Comment by miaoyq** [comment]
>>"protobufObjectToJSON(status)" handles resp.Status, and "json.Marshal" handles resp.Info
>
>LGTM. @Random-Liu do you have any other concerns?

@Random-Liu @feiskyer  Updated, also related to #166.

File: cmd/crictl/util.go [hunk 2]
```diff
@@ -178,8 +180,18 @@ func outputYAML(v interface{}) error {
...
 
-func outputStatusInfo(status interface{}, info map[string]string, format string) error {
-	statusByte, err := json.Marshal(status)
+func protobufObjectToJSON(obj proto.Message) (string, error) {
+	jsonpbMarshaler := jsonpb.Marshaler{EmitDefaults: true, Indent: "  "}
+	marshaledJSON, err := jsonpbMarshaler.MarshalToString(obj)
+	if err != nil {
+		return "", err
+	}
+
```
Comment (Random-Liu) [inline cmd/crictl/util.go]: nit: unnecessary empty line.

**Comment by Random-Liu** [comment]
@miaoyq For some other commands, we also support yaml/json output. We should also use jsonpb for those cases. I'm fine with doing that in another PR, but please make sure the issue is open.

File: cmd/crictl/util.go (hunk 2 above)
Comment (miaoyq) [reply cmd/crictl/util.go]: Done.

**Comment by miaoyq** [comment]
>@miaoyq For some other commands, we also support yaml/json output. We should also use jsonpb for those cases. I'm fine with doing that in another PR, but please make sure the issue is open.

Will do.

**Pull request merged**